local-volume/provisioner/pkg/common/common.go

```diff
@@ -122,13 +126,25 @@ func CreateLocalPVSpec(config *LocalPVConfig) *v1.PersistentVolume {
 	}
 }
 
-// GetVolumeConfigFromConfigMap gets volume configuration from given configmap,
-func GetVolumeConfigFromConfigMap(client *kubernetes.Clientset, namespace, name string) (map[string]MountConfig, error) {
-	configMap, err := client.CoreV1().ConfigMaps(namespace).Get(name, metav1.GetOptions{})
+// GetVolumeConfigFromConfigMap gets volume configuration from the mounted configmap volume,
+func GetVolumeConfigFromConfigMap() (map[string]MountConfig, error) {
+	volumes := make(map[string]string)
+	files, err := ioutil.ReadDir(provisionerConfigPath)
 	if err != nil {
 		return nil, err
 	}
-	return ConfigMapDataToVolumeConfig(configMap.Data)
+	for _, file := range files {
+		if !file.IsDir() {
+			b, err := ioutil.ReadFile(path.Join(provisionerConfigPath, file.Name()))
+			fmt.Printf("><><><> reading file: %s <><><><\n", path.Join(provisionerConfigPath, file.Name()))
```

Comment:
Instead of using Printf, let's use glog.

Reply:
@ianchakeres thank you for your comment, these are leftover from my debugging I was planning to remove them completely before the final cut.
I was wondering if you are ok in general with approach I took.

Reply:
I think the approach is ok, and it conforms to the current config format.

Reply:
The current configuration assumes that all the files (aka keys inside data:) are storageClassNames. I wonder if we should update the config format to allow for additional configuration options that do not conform to this top-level key structure. We could accomplish this by putting the existing json config as a value to a new single key (e.g. localVolumeClasses), and that would result in processing of a single file here. @msau42 


@msau42

Reply:
@ianchakeres I see what you mean.  The function I added does not decode json, but just prep data for the already existing json decoding function. I would like to ask if it is possible to keep  your suggestion separate from this PR.  The function which is used to parse encoded json is called I think from other places and changing config format will require a bit bigger refactoring.  Would you agree?

Reply:
@sbezverk - I agree.

Reply:
Agree as well.  I think it may be more user-friendly to not use encoded json as the configmap data format.